# Import Dependencies

In [4]:
import numpy as np
import pandas as pd

pd.set_option('display.max_columns', 500)
pd.set_option('display.max_rows', 500)

# Read in Data

In [5]:
# Columns to keep for the data we want to use
keep_cols= ["leaidC","leaname","fips","stateabb","grade","year","subject",
          "mn_all","mn_asn","mn_blk","mn_fem","mn_hsp","mn_mal","mn_wht",
          "mn_wag","mn_wbg","mn_whg","mn_mfg"
]

In [2]:
# long_CS= pd.read_csv("resources\datasets\SEDA_geodist_long_CS_v21.csv")

In [3]:
# long_gcs= pd.read_csv("resources\datasets\SEDA_geodist_long_GCS_v21.csv")

In [6]:
# This is the dataset we'll be using
long_naep= pd.read_csv("resources\datasets\SEDA_geodist_long_NAEP_v21.csv")
long_naep= long_naep.loc[:,keep_cols]

In [7]:
# long_state= pd.read_csv("resources\datasets\SEDA_geodist_long_State_v21.csv")

In [8]:
# pool_cs= pd.read_csv("resources\datasets\SEDA_geodist_pool_CS_v21.csv")

In [9]:
# poolsub_cs= pd.read_csv("resources\datasets\SEDA_geodist_poolsub_CS_v21.csv")

# Columns in the data

In [10]:
# print(len(long_CS.columns))
# print(len(long_gcs.columns))
# print(len(long_naep.columns))
# print(len(long_state.columns))
# print(len(pool_cs.columns))
# print(len(poolsub_cs.columns))

In [11]:
# columns_df= pd.DataFrame({
#     "long_CS": pd.Series(long_CS.columns),
#     "long_gcs": pd.Series(long_gcs.columns),
#     "long_naep": pd.Series(long_naep.columns),
#     "long_state": pd.Series(long_state.columns),
#     "pool_cs": pd.Series(pool_cs.columns),
#     "poolsub_cs": pd.Series(poolsub_cs.columns)
# })

In [12]:
# columns_df

# Off with their heads!

In [13]:
# long_CS.head()

In [14]:
# long_gcs.head()

In [16]:
long_naep.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

      mn_all  mn_asn     mn_blk  mn_fem  mn_hsp     mn_mal  mn_wht  mn_wag  \
0  209.40759     NaN        NaN     NaN     NaN  207.21269     NaN     NaN   
1  217.34308     NaN  208.98753     NaN     NaN  215.39435     NaN     NaN   
2  230.81302     NaN        NaN     NaN     NaN  229.81973     NaN     NaN   
3  221.81458     NaN        NaN     NaN     NaN  223.95609     NaN     NaN   
4  227.06160     NaN        NaN     NaN     NaN  224.74576     NaN     NaN   

   mn_wbg  mn_whg  mn_mfg  
0     NaN     NaN     NaN  
1     NaN     NaN     NaN  
2     NaN     NaN     NaN  
3     NaN     NaN     NaN  
4     NaN     NaN     NaN

In [17]:
# long_state.head()

In [18]:
# pool_cs.head()

In [19]:
# poolsub_cs.head()

# Describe the datasets

In [21]:
# desc_long_CS= long_CS.describe()
# desc_long_gcs= long_gcs.describe()
desc_long_naep= long_naep.describe()
# desc_long_state= long_state.describe()
# desc_pool_cs= pool_cs.describe()
# desc_poolsub_cs= poolsub_cs.describe()

In [22]:
# desc_long_CS

In [23]:
# desc_long_gcs

In [25]:
desc_long_naep

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

              mn_all        mn_asn         mn_blk         mn_fem  \
count  825814.000000  85523.000000  183171.000000  684574.000000   
mean      245.368960    264.213928     228.758834     248.236700   
std        24.817281     30.265167      22.675072      24.501244   
min       114.207150     99.107666     129.180240     123.957090   
25%       228.196378    243.636095     213.230470     230.850810   
50%       245.202560    263.100400     229.219020     247.870575   
75%       262.570257    283.998410     244.799475     265.586045   
max       375.458440    441.867650     314.946560     403.577090   

              mn_hsp         mn_mal         mn_wht        mn_wag  \
count  222124.000000  695482.000000  733460.000000  84024.000000   
mean      233.040101     243.568544     250.108341     -7.432532   
std        24.205590      25.659113      23.724225     14.264807   
min       132.774960     113.350510     130.434510   -131.135770   
25%       216.558828     226.164378     233.195263    -16.002594   
50%       233.525605     243.450965     249.758835     -8.062675   
75%       249.974620     260.939640     266.682310      0.369026   
max       336.939790     377.005740     376.761720    137.154270   

              mn_wbg         mn_whg         mn_mfg  
count  164413.000000  196243.000000  658277.000000  
mean       20.692708      16.047852      -4.591977  
std         9.994392      10.275893       8.664576  
min       -25.843262     -43.132675    -106.446230  
25%        14.246857       9.267326      -9.915894  
50%        20.054749      15.276093      -4.178528  
75%        26.478058      22.158958       1.042725  
max       147.544370     153.679550      58.579193

In [26]:
# desc_long_state

In [27]:
# desc_pool_cs

In [28]:
# desc_poolsub_cs

# Trying to make sense of the data

## NAEP

In [29]:
desc_long_naep

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

              mn_all        mn_asn         mn_blk         mn_fem  \
count  825814.000000  85523.000000  183171.000000  684574.000000   
mean      245.368960    264.213928     228.758834     248.236700   
std        24.817281     30.265167      22.675072      24.501244   
min       114.207150     99.107666     129.180240     123.957090   
25%       228.196378    243.636095     213.230470     230.850810   
50%       245.202560    263.100400     229.219020     247.870575   
75%       262.570257    283.998410     244.799475     265.586045   
max       375.458440    441.867650     314.946560     403.577090   

              mn_hsp         mn_mal         mn_wht        mn_wag  \
count  222124.000000  695482.000000  733460.000000  84024.000000   
mean      233.040101     243.568544     250.108341     -7.432532   
std        24.205590      25.659113      23.724225     14.264807   
min       132.774960     113.350510     130.434510   -131.135770   
25%       216.558828     226.164378     233.195263    -16.002594   
50%       233.525605     243.450965     249.758835     -8.062675   
75%       249.974620     260.939640     266.682310      0.369026   
max       336.939790     377.005740     376.761720    137.154270   

              mn_wbg         mn_whg         mn_mfg  
count  164413.000000  196243.000000  658277.000000  
mean       20.692708      16.047852      -4.591977  
std         9.994392      10.275893       8.664576  
min       -25.843262     -43.132675    -106.446230  
25%        14.246857       9.267326      -9.915894  
50%        20.054749      15.276093      -4.178528  
75%        26.478058      22.158958       1.042725  
max       147.544370     153.679550      58.579193

The NAEP score is derived from the [National Assessment of Educational Progress](https://nces.ed.gov/nationsreportcard/ "NAEP Nations Report Card"). The test is on a 0-500 scale, which we can see reflected in the means of all the schools *(mn_all)*.

In [30]:
print(f'Minimum AVERAGE score for all schools: {desc_long_naep.loc["min","mn_all"]}')
print(f'Maximum AVERAGE score for all schools: {desc_long_naep.loc["max","mn_all"]}')

Minimum AVERAGE score for all schools: 114.20715
Maximum AVERAGE score for all schools: 375.45844


The above shows that the AVERAGE maximum score for any school is 375.5. This isn't to say that the highest score that a student received on the test is that number. Rather, that school could've had someone score the full 500 and another student who scored in the double digits. This is important to keep in mind while parsing through the data.

The other measures use various statistical techniques (HETOP and HOMOP, for example) to derive their normalized values, which I did not want to dig much deeper into. I believe using the NAEP score should be sufficient for the purposes of this data analysis.

Another interesting story comes from the "gaps" (disparity in average scores between sub-groups within a year/class/subject)

In [31]:
# Columns which contain values for the gaps between scores of:
# White v Asian students
#       v Black students
#       v Hispanic students
# Male v Female students

gap_cols= ["mn_wag","mn_wbg","mn_whg","mn_mfg"]

desc_long_naep.loc[:,gap_cols]

mn_wag         mn_wbg         mn_whg         mn_mfg
count  84024.000000  164413.000000  196243.000000  658277.000000
mean      -7.432532      20.692708      16.047852      -4.591977
std       14.264807       9.994392      10.275893       8.664576
min     -131.135770     -25.843262     -43.132675    -106.446230
25%      -16.002594      14.246857       9.267326      -9.915894
50%       -8.062675      20.054749      15.276093      -4.178528
75%        0.369026      26.478058      22.158958       1.042725
max      137.154270     147.544370     153.679550      58.579193

Looking at the min and max from the *mn_wag* column, it looks like there is a vast disparity between the scores of White students and Asian students (min: -131.1, max: 137.2). In fact, this suggests the opposite.

These scores are derived by subtracting the first subgroup ("W") from the second ("A"). So it appears that in at least one school, Asian student scores are __on average higher__ than their White peers by 131 points. The converse is true given the max value as well.

Armed with this information, we can see that there does indeed appear to be a disparity between White and Black students' as well as White and Hispanic students' scores. The same holds for Male v Female scores, with the disparity favoring Female students.

# Produce Datasets

Derived from the NAEP dataset

* Demographic Inclusive
    * ELA
    * Math
* Demographic Exclusive
    * ELA
    * Math

### Demo ELA

In [32]:
demo_ela= long_naep.loc[long_naep["subject"] == "ela"].dropna(how="any")

In [33]:
demo_ela.head()

leaidC      leaname  fips stateabb  grade  year subject     mn_all  \
172  100007  Hoover City     1       AL      3  2009     ela  214.54395   
176  100007  Hoover City     1       AL      3  2011     ela  217.66377   
178  100007  Hoover City     1       AL      3  2012     ela  215.21539   
180  100007  Hoover City     1       AL      3  2013     ela  213.73804   
182  100007  Hoover City     1       AL      3  2014     ela  219.03290   

        mn_asn     mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
172  220.27069  190.93071  217.31458  195.04224  212.36569  224.37381   
176  222.10507  193.69196  219.95566  189.55688  214.78072  231.69139   
178  220.94724  191.52635  219.43379  185.46619  211.59489  226.93578   
180  220.89682  196.55606  215.47179  184.31003  211.50256  224.43622   
182  229.80255  197.46252  222.84422  190.68884  215.92966  228.89838   

       mn_wag     mn_wbg     mn_whg    mn_mfg  
172  4.103119  33.443100  29.331573 -4.948883  
176  9.586319  37.999435  42.134506 -5.174942  
178  5.988541  35.409424  41.469589 -7.838898  
180  3.539398  27.880157  40.126190 -3.969223  
182 -0.904175  31.435852  38.209534 -6.914566

### Demo Math

In [34]:
demo_math= long_naep.loc[long_naep["subject"] == "math"].dropna(how="any")

In [35]:
demo_math.head()

leaidC      leaname  fips stateabb  grade  year subject     mn_all  \
173  100007  Hoover City     1       AL      3  2009    math  222.00957   
177  100007  Hoover City     1       AL      3  2011    math  223.60043   
179  100007  Hoover City     1       AL      3  2012    math  223.30418   
181  100007  Hoover City     1       AL      3  2013    math  226.40253   
183  100007  Hoover City     1       AL      3  2014    math  230.91217   

        mn_asn     mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
173  228.68329  205.59877  220.91368  200.90639  222.98395  228.69203   
177  231.37659  207.22340  222.21732  208.76343  225.15512  231.22810   
179  234.07726  205.46187  223.13036  201.37108  223.36121  229.63614   
181  245.31400  213.29327  225.05655  210.97031  227.76344  231.85593   
183  244.89435  216.53716  230.94514  218.20084  230.68436  236.18112   

        mn_wag     mn_wbg     mn_whg    mn_mfg  
173   0.008743  23.093262  27.785645  2.070267  
177  -0.148483  24.004700  22.464676  2.937805  
179  -4.441116  24.174271  28.265060  0.230850  
181 -13.458069  18.562653  20.885620  2.706894  
183  -8.713226  19.643967  17.980286 -0.260788

### All ELA

In [36]:
all_ela= long_naep.loc[long_naep["subject"] == "ela"]

In [37]:
all_ela.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   
6  100005        Albertville City     1       AL      3  2009     ela   
8  100005        Albertville City     1       AL      3  2010     ela   

      mn_all  mn_asn  mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
0  209.40759     NaN     NaN        NaN        NaN  207.21269        NaN   
2  230.81302     NaN     NaN        NaN        NaN  229.81973        NaN   
4  227.06160     NaN     NaN        NaN        NaN  224.74576        NaN   
6  204.40581     NaN     NaN  206.97821  195.20105  201.54659  211.53754   
8  206.01045     NaN     NaN        NaN  196.38988        NaN  211.12666   

   mn_wag  mn_wbg     mn_whg    mn_mfg  
0     NaN     NaN        NaN       NaN  
2     NaN     NaN        NaN       NaN  
4     NaN     NaN        NaN       NaN  
6     NaN     NaN  16.336487 -5.431625  
8     NaN     NaN  14.736786       NaN

### All Math

In [38]:
all_math= long_naep.loc[long_naep["subject"] == "math"]

In [39]:
all_math.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
1  100002  Alabama Youth Services     1       AL      8  2009    math   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
5  100002  Alabama Youth Services     1       AL      8  2012    math   
7  100005        Albertville City     1       AL      3  2009    math   
9  100005        Albertville City     1       AL      3  2010    math   

      mn_all  mn_asn     mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
1  217.34308     NaN  208.98753        NaN        NaN  215.39435        NaN   
3  221.81458     NaN        NaN        NaN        NaN  223.95609        NaN   
5  229.63074     NaN        NaN        NaN        NaN  229.88904        NaN   
7  216.63632     NaN        NaN  216.87672  211.63123  215.30283  220.40680   
9  228.81721     NaN        NaN        NaN  223.30786        NaN  233.30249   

   mn_wag  mn_wbg    mn_whg    mn_mfg  
1     NaN     NaN       NaN       NaN  
3     NaN     NaN       NaN       NaN  
5     NaN     NaN       NaN       NaN  
7     NaN     NaN  8.775574 -1.573898  
9     NaN     NaN  9.994629       NaN

## Save to CSV

In [40]:
# demo_ela.to_csv(r"resources\datasets\demo_ela.csv")
# demo_math.to_csv(r"resources\datasets\demo_math.csv")
# all_ela.to_csv(r"resources\datasets\all_ela.csv")
# all_math.to_csv(r"resources\datasets\all_math.csv")

# Investigating data integrity at different levels

In [41]:
print(len(demo_ela))
print(len(demo_math))
print(len(all_ela))
print(len(all_math))

27516
26423
424550
401266


In [42]:
col= "stateabb"

print(len(demo_ela[col].value_counts()))
print(len(demo_math[col].value_counts()))
print(len(all_ela[col].value_counts()))
print(len(all_math[col].value_counts()))

50
50
51
51


In [43]:
long_naep.loc[long_naep["stateabb"] == "HI"]

leaidC                         leaname  fips stateabb  grade  year  \
126610  1500030  Hawaii Department Of Education    15       HI      3  2009   
126611  1500030  Hawaii Department Of Education    15       HI      3  2009   
126612  1500030  Hawaii Department Of Education    15       HI      3  2010   
126613  1500030  Hawaii Department Of Education    15       HI      3  2010   
126614  1500030  Hawaii Department Of Education    15       HI      3  2011   
126615  1500030  Hawaii Department Of Education    15       HI      3  2011   
126616  1500030  Hawaii Department Of Education    15       HI      3  2012   
126617  1500030  Hawaii Department Of Education    15       HI      3  2012   
126618  1500030  Hawaii Department Of Education    15       HI      3  2013   
126619  1500030  Hawaii Department Of Education    15       HI      3  2013   
126620  1500030  Hawaii Department Of Education    15       HI      3  2014   
126621  1500030  Hawaii Department Of Education    15       HI      3  2014   
126622  1500030  Hawaii Department Of Education    15       HI      3  2015   
126623  1500030  Hawaii Department Of Education    15       HI      3  2015   
126624  1500030  Hawaii Department Of Education    15       HI      4  2009   
126625  1500030  Hawaii Department Of Education    15       HI      4  2009   
126626  1500030  Hawaii Department Of Education    15       HI      4  2010   
126627  1500030  Hawaii Department Of Education    15       HI      4  2010   
126628  1500030  Hawaii Department Of Education    15       HI      4  2011   
126629  1500030  Hawaii Department Of Education    15       HI      4  2011   
126630  1500030  Hawaii Department Of Education    15       HI      4  2012   
126631  1500030  Hawaii Department Of Education    15       HI      4  2012   
126632  1500030  Hawaii Department Of Education    15       HI      4  2013   
126633  1500030  Hawaii Department Of Education    15       HI      4  2013   
126634  1500030  Hawaii Department Of Education    15       HI      4  2014   
126635  1500030  Hawaii Department Of Education    15       HI      4  2014   
126636  1500030  Hawaii Department Of Education    15       HI      4  2015   
126637  1500030  Hawaii Department Of Education    15       HI      4  2015   
126638  1500030  Hawaii Department Of Education    15       HI      5  2009   
126639  1500030  Hawaii Department Of Education    15       HI      5  2009   
126640  1500030  Hawaii Department Of Education    15       HI      5  2010   
126641  1500030  Hawaii Department Of Education    15       HI      5  2010   
126642  1500030  Hawaii Department Of Education    15       HI      5  2011   
126643  1500030  Hawaii Department Of Education    15       HI      5  2011   
126644  1500030  Hawaii Department Of Education    15       HI      5  2012   
126645  1500030  Hawaii Department Of Education    15       HI      5  2012   
126646  1500030  Hawaii Department Of Education    15       HI      5  2013   
126647  1500030  Hawaii Department Of Education    15       HI      5  2013   
126648  1500030  Hawaii Department Of Education    15       HI      5  2014   
126649  1500030  Hawaii Department Of Education    15       HI      5  2014   
126650  1500030  Hawaii Department Of Education    15       HI      5  2015   
126651  1500030  Hawaii Department Of Education    15       HI      5  2015   
126652  1500030  Hawaii Department Of Education    15       HI      6  2009   
126653  1500030  Hawaii Department Of Education    15       HI      6  2009   
126654  1500030  Hawaii Department Of Education    15       HI      6  2010   
126655  1500030  Hawaii Department Of Education    15       HI      6  2010   
126656  1500030  Hawaii Department Of Education    15       HI      6  2011   
126657  1500030  Hawaii Department Of Education    15       HI      6  2011   
126658  1500030  Hawaii Department Of Education    15       HI      6  2012   
126659  1500030  Hawaii Department Of Education    15      

In [44]:
col= "stateabb"

stateabb_df= pd.DataFrame({
    "demo_ela": demo_ela[col].value_counts(),
    "demo_math": demo_math[col].value_counts(),
    "all_ela": all_ela[col].value_counts(),
    "all_math": all_math[col].value_counts()
})

# print(demo_ela[col].value_counts())
# print(demo_math[col].value_counts())
# print(all_ela[col].value_counts())
# print(all_math[col].value_counts())

stateabb_df

demo_ela  demo_math  all_ela  all_math
AK      96.0      102.0     1302      1299
AL     227.0      239.0     5489      5500
AR     206.0      206.0     9567      8859
AZ    1008.0     1011.0     6021      6035
CA    4615.0     3335.0    26362     19063
CO     250.0      267.0     2342      2277
CT     590.0      630.0     5375      5367
DC      33.0       40.0       41        41
DE     196.0      224.0      649       650
FL     887.0      781.0     2865      2471
GA     737.0      988.0     7517      7518
HI      42.0       42.0       42        42
IA     291.0      316.0    12627     12618
ID      61.0       77.0     3075      3084
IL    1201.0     1241.0    28006     28065
IN     378.0      494.0    12037     12045
KS     195.0      258.0     8000      8040
KY     180.0      233.0     7142      7142
LA     323.0      366.0     3042      3042
MA     628.0      650.0    10415     10439
MD     453.0      518.0      922       923
ME      25.0       33.0     4454      4369
MI     563.0      733.0    20686     20722
MN    1014.0     1035.0    12699     12711
MO     587.0      604.0    16083     14937
MS     124.0      124.0     5910      5901
MT      12.0       13.0     3056      3076
NC     794.0      868.0     4840      4840
ND      22.0       29.0     2386      2381
NE     124.0      118.0     5599      4666
NH      44.0       47.0     4613      4613
NJ    1753.0     1776.0    16327     16328
NM     129.0      129.0     2797      2790
NV      61.0       62.0      457       459
NY     966.0      999.0    20331     20142
OH     531.0      565.0    25224     24623
OK     484.0      475.0    15034     14374
OR     251.0      291.0     5387      5024
PA     910.0      982.0    20845     20853
RI     113.0      122.0     1259      1277
SC     537.0      570.0     3523      3523
SD      27.0       35.0     3346      3350
TN     378.0      436.0     5584      5457
TX    2793.0     2274.0    34805     28211
UT     314.0      278.0     1601      1401
VA     885.0      318.0     5525      1835
VT       NaN        NaN     3720      3721
WA     848.0      857.0     6459      6457
WI     609.0      613.0    15823     15816
WV      15.0       13.0     2324      2048
WY       6.0        6.0     1015       841

In [45]:
col= "grade"

print(demo_ela[col].value_counts())
print(demo_math[col].value_counts())
print(all_ela[col].value_counts())
print(all_math[col].value_counts())

5    4673
4    4640
3    4614
6    4596
7    4499
8    4494
Name: grade, dtype: int64
3    4942
4    4926
5    4753
6    4653
7    3666
8    3483
Name: grade, dtype: int64
5    71430
4    71310
3    71061
6    70888
7    69956
8    69905
Name: grade, dtype: int64
4    70827
3    70788
5    70074
6    69520
7    61764
8    58293
Name: grade, dtype: int64


In [46]:
col= "leaname"

print(len(demo_ela[col].value_counts()))
print(len(demo_math[col].value_counts()))
print(len(all_ela[col].value_counts()))
print(len(all_math[col].value_counts()))

1202
1210
11723
11699


In [47]:
col= "year"

print(demo_ela[col].value_counts())
print(demo_math[col].value_counts())
print(all_ela[col].value_counts())
print(all_math[col].value_counts())

2012    4933
2013    4417
2011    4241
2010    4124
2009    3859
2015    3170
2014    2772
Name: year, dtype: int64
2012    4367
2013    4234
2011    4074
2010    3919
2009    3563
2015    3446
2014    2820
Name: year, dtype: int64
2012    64590
2011    64478
2010    64310
2013    64018
2009    63786
2015    52945
2014    50423
Name: year, dtype: int64
2011    62424
2009    61490
2010    61103
2012    60618
2013    59421
2015    49473
2014    46737
Name: year, dtype: int64


# New DF restrictions
Restrict data to all schools in 2015.

In [48]:
ela2015_df= long_naep.loc[(long_naep["subject"] == "ela") & (long_naep["year"] == 2015)]
math2015_df= long_naep.loc[(long_naep["subject"] == "math") & (long_naep["year"] == 2015)]

In [49]:
print(len(ela2015_df))
print(len(math2015_df))

52945
49473


In [50]:
print(len(ela2015_df["leaname"].value_counts()))
print(len(math2015_df["leaname"].value_counts()))

9512
9464


In [51]:
ela2015_df.to_csv("resources/ignore_datasets/ela_2015.csv")
math2015_df.to_csv("resources/ignore_datasets/math_2015.csv")

In [52]:
len(ela2015_df.dropna(how= 'any'))

3170

# New DF restrictions (2013)
Restrict data to all schools in 2013.

In [100]:
ela2013_df= long_naep.loc[(long_naep["subject"] == "ela") & (long_naep["year"] == 2013)]
math2013_df= long_naep.loc[(long_naep["subject"] == "math") & (long_naep["year"] == 2013)]

ela2013_df.fillna(ela2013_df.mean(), inplace= True)
math2013_df.fillna(math2013_df.mean(), inplace= True)

C:\Users\Overlord\Anaconda3\envs\PythonData\lib\site-packages\pandas\core\generic.py:5434: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


In [101]:
print(len(ela2013_df))
print(len(math2013_df))

64018
59421


In [102]:
print(len(ela2013_df["leaname"].value_counts()))
print(len(math2013_df["leaname"].value_counts()))

11164
11078


In [103]:
ela2013_df.to_csv("resources/data/ela_2013.csv")
math2013_df.to_csv("resources/data/math_2013.csv")

# Aggregated csv's

In [105]:
ela2013_agg_df= ela2013_df.groupby(["grade","stateabb"]).mean()
math2013_agg_df= math2013_df.groupby(["grade","stateabb"]).mean()

ela2013_agg_df.reset_index(inplace= True)
math2013_agg_df.reset_index(inplace= True)

ela2013_agg_df.fillna(ela2013_agg_df.mean(), inplace= True)
math2013_agg_df.fillna(math2013_agg_df.mean(), inplace= True)

In [106]:
ela2013_agg_df
math2013_agg_df

grade stateabb        leaidC  fips    year      mn_all      mn_asn  \
0        3       AK  2.003362e+05   2.0  2013.0  215.387929  265.880157   
1        3       AL  1.016995e+05   1.0  2013.0  223.414471  273.587803   
2        3       AR  5.068369e+05   5.0  2013.0  229.937960  274.917467   
3        3       AZ  4.049385e+05   4.0  2013.0  225.432389  270.081678   
4        3       CA  6.215712e+05   6.0  2013.0  221.024373  261.558090   
5        3       CO  8.044917e+05   8.0  2013.0  235.661912  270.992080   
6        3       CT  9.027074e+05   9.0  2013.0  236.561988  270.410310   
7        3       DC  1.100030e+06  11.0  2013.0  218.800000  253.059840   
8        3       DE  1.000797e+06  10.0  2013.0  233.417264  265.891313   
9        3       FL  1.201063e+06  12.0  2013.0  230.963703  265.765781   
10       3       GA  1.302877e+06  13.0  2013.0  228.199217  272.252629   
11       3       HI  1.500030e+06  15.0  2013.0  233.125000  231.768540   
12       3       IA  1.916333e+06  19.0  2013.0  236.449871  274.673888   
13       3       ID  1.601712e+06  16.0  2013.0  226.314609  273.340658   
14       3       IL  1.721574e+06  17.0  2013.0  229.219375  272.857136   
15       3       IN  1.806556e+06  18.0  2013.0  239.957003  274.332090   
16       3       KS  2.007558e+06  20.0  2013.0  235.622002  274.147291   
17       3       KY  2.102928e+06  21.0  2013.0  229.334897  274.465061   
18       3       LA  2.200941e+06  22.0  2013.0  219.033805  269.457946   
19       3       MA  2.507233e+06  25.0  2013.0  243.116533  272.373239   
20       3       MD  2.400375e+06  24.0  2013.0  235.990047  263.793323   
21       3       ME  2.311388e+06  23.0  2013.0  233.905871  275.087910   
22       3       MI  2.619419e+06  26.0  2013.0  225.072128  273.297222   
23       3       MN  2.718233e+06  27.0  2013.0  243.072219  272.343899   
24       3       MO  2.918729e+06  29.0  2013.0  228.410614  273.935538   
25       3       MS  2.802604e+06  28.0  2013.0  218.346291  274.848732   
26       3       MT  3.013460e+06  30.0  2013.0  231.008903  275.240828   
27       3       NC  3.702454e+06  37.0  2013.0  231.468814  270.919298   
28       3       ND  3.808361e+06  38.0  2013.0  234.004692  273.661903   
29       3       NE  3.137168e+06  31.0  2013.0  230.600789  275.024199   
30       3       NH  3.304073e+06  33.0  2013.0  243.262852  274.888297   
31       3       NJ  3.409458e+06  34.0  2013.0  236.051181  270.589848   
32       3       NM  3.501363e+06  35.0  2013.0  219.684948  274.191266   
33       3       NV  3.200274e+06  32.0  2013.0  222.616570  270.269900   
34       3       NY  3.616523e+06  36.0  2013.0  229.588124  272.107875   
35       3       OH  3.904990e+06  39.0  2013.0  236.497020  274.645680   
36       3       OK  4.017670e+06  40.0  2013.0  230.206755  274.207326   
37       3       OR  4.106766e+06  41.0  2013.0  225.611770  271.860677   
38       3       PA  4.214602e+06  42.0  2013.0  234.198158  273.416422   
39       3       RI  4.400612e+06  44.0  2013.0  236.961268  270.500511   
40       3       SC  4.502364e+06  45.0  2013.0  221.912302  270.248777   
41       3       SD  4.638218e+06  46.0  2013.0  230.143588  274.506602   
42       3       TN  4.702307e+06  47.0  2013.0  229.182432  272.835005   
43       3       TX  4.826496e+06  48.0  2013.0  228.509061  273.222635   
44       3       UT  4.900604e+06  49.0  2013.0  230.968194  262.430038   
45       3       VA  5.102123e+06  51.0  2013.0  230.956539  270.326841   
46       3       VT  5.006416e+06  50.0  2013.0  236.078290  275.165597   
47       3       WA  5.305070e+06  53.0  2013.0  230.489500  268.677699   
48       3       WI  5.508411e+06  55.0  2013.0  234.525512  272.541533   
49       3       WV  5.400855e+06  54.0  2013.0  226.637259  275.022517   
50       4       AK  2.003715e+05   2.0  2013.0  229.429844  268.177147   
51       4       AL  1.016995e+05   1.0  2013.0  230.894551  274.848944   
52       4       AR  5.068455e

In [107]:
ela2013_agg_df.to_csv("resources/data/ela_2013_agg.csv")
math2013_agg_df.to_csv("resources/data/math_2013_agg.csv")

# Aggregations by state

In [108]:
ela_2013_state_agg= ela2013_agg_df.groupby("stateabb").mean()
math_2013_state_agg= math2013_agg_df.groupby("stateabb").mean()

ela_2013_state_agg.fillna(ela_2013_state_agg.mean(), inplace= True)
math_2013_state_agg.fillna(math_2013_state_agg.mean(), inplace= True)

In [109]:
ela_2013_state_agg.to_csv("resources/data/ela_2013_state_agg.csv")
math_2013_state_agg.to_csv("resources/data/math_2013_state_agg.csv")

In [110]:
ela_2013_state_agg.head()

grade         leaidC  fips    year      mn_all      mn_asn  \
stateabb                                                               
AK          5.5  200358.615099   2.0  2013.0  219.290018  249.961407   
AL          5.5  101699.485075   1.0  2013.0  230.312728  254.951118   
AR          5.5  506810.869300   5.0  2013.0  232.571712  254.629608   
AZ          5.5  404911.136751   4.0  2013.0  224.005371  252.915935   
CA          5.5  621559.519063   6.0  2013.0  229.612175  252.650918   

              mn_blk      mn_fem      mn_hsp      mn_mal      mn_wht  \
stateabb                                                               
AK        220.915841  230.938361  225.243979  223.123724  243.287805   
AL        217.208742  235.369524  223.419625  225.448379  242.473163   
AR        220.111172  242.947207  224.849475  225.743276  237.582815   
AZ        220.236186  231.796824  221.843153  222.113431  239.584792   
CA        220.931011  235.485396  220.753306  227.256810  241.424606   

            mn_wag     mn_wbg     mn_whg     mn_mfg  
stateabb                                             
AK       -1.076326  21.763858  17.134853  -9.105615  
AL       -4.625826  24.500572  19.472142  -9.808630  
AR       -4.776471  21.763652  17.600752 -15.901997  
AZ       -4.300385  20.985112  16.693415  -9.457758  
CA       -5.027218  21.275253  18.940305  -7.998839